projects/ion/src/lib/checkbox/checkbox.component.ts

```diff
@@ -0,0 +1,31 @@
+import { Component, Input, Output, EventEmitter } from '@angular/core';
+
+export interface CheckBoxProps {
+  disabled?: boolean;
+  checked?: boolean;
+  indeterminate?: boolean;
+  ionClick?: EventEmitter<CheckBoxEvent>;
+}
+
+interface CheckBoxEvent {
+  checked?: boolean;
+}
+
+@Component({
+  selector: 'ion-checkbox',
+  templateUrl: './checkbox.component.html',
+  styleUrls: ['./checkbox.component.scss'],
+})
+export class CheckboxComponent {
+  @Input() public disabled?: boolean = false;
+  @Input() public checked?: boolean = false;
+  @Input() public indeterminate?: boolean = false;
```

Comment:
```suggestion
  @Input() public disabled? = false;
  @Input() public checked? = false;
  @Input() public indeterminate? = false;
```
Typescript can infer the types just by the value, it's a good practice when values are number, string or boolean, make the code easier to read 😄